includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -1120,61 +1125,68 @@ protected function do_transaction( $order ) {
+					esc_html__( '%1$s %2$s %3$s Approved for an amount of %4$s', 'woocommerce-square' ),
+					$this->get_method_title(),
+					wc_square()->get_settings_handler()->is_sandbox() ? esc_html_x( 'Test', 'noun, software environment', 'woocommerce-square' ) : '',
+					'APPROVED' === $response->get_payment()->getStatus() ? esc_html_x( 'Authorization', 'Cash App transaction type', 'woocommerce-square' ) : esc_html_x( 'Charge', 'noun, Cash App transaction type', 'woocommerce-square' ),
+					wc_price( Money_Utility::cents_to_float( $payment->getTotalMoney()->getAmount(), $order->get_currency() ) )
+				);
+
+				// adds the transaction id (if any) to the order note
+				if ( $response->get_transaction_id() ) {
+					/* translators: Placeholders: %s - transaction ID */
+					$message .= ' ' . sprintf( esc_html__( '(Transaction ID %s)', 'woocommerce-square' ), $response->get_transaction_id() );
+				}
 
-			// adds the transaction id (if any) to the order note
-			if ( $response->get_transaction_id() ) {
-				/* translators: Placeholders: %s - transaction ID */
-				$message .= ' ' . sprintf( esc_html__( '(Transaction ID %s)', 'woocommerce-square' ), $response->get_transaction_id() );
+				/**
+				 * Direct Gateway Credit Card Transaction Approved Order Note Filter.
+				 *
+				 * Allow actors to modify the order note added when a Credit Card transaction
+				 * is approved.
+				 *
+				 * @since 4.5.0
+				 *
+				 * @param string $message order note
+				 * @param \WC_Order $order order object
+				 * @param \WooCommerce\Square\Gateway\API\Response $response transaction response
+				 * @param Cash_App_Pay_Gateway $this instance
+				 */
+				$message = apply_filters( 'wc_payment_gateway_' . $this->get_id() . '_transaction_approved_order_note', $message, $order, $response, $this );
+
+				$this->update_order_meta( $order, 'is_tender_type_cash_app_wallet', true );
+
+				$order->add_order_note( $message );
 			}
 
-			/**
-			 * Direct Gateway Credit Card Transaction Approved Order Note Filter.
-			 *
-			 * Allow actors to modify the order note added when a Credit Card transaction
-			 * is approved.
-			 *
-			 * @since 4.5.0
-			 *
-			 * @param string $message order note
-			 * @param \WC_Order $order order object
-			 * @param \WooCommerce\Square\Gateway\API\Response $response transaction response
-			 * @param Cash_App_Pay_Gateway $this instance
-			 */
-			$message = apply_filters( 'wc_payment_gateway_' . $this->get_id() . '_transaction_approved_order_note', $message, $order, $response, $this );
-
-			$this->update_order_meta( $order, 'is_tender_type_cash_app_wallet', true );
-
-			$order->add_order_note( $message );
+			return $response;
+		} catch ( \Exception $e ) {
+			Performance_Logger::end( 'payment_transaction', $this->get_plugin(), true );
+			throw $e;
```

Comment:
And here

Reply:
See https://github.com/woocommerce/woocommerce-square/pull/279#discussion_r1987446536